Enable munin-node-c to build and run on macOS/Darwin

smokris opened the pull request:
A couple small changes that enable `munin-node-c` to build and run on macOS/Darwin.  Tested on OS X 10.10, macOS 10.13, and macOS 10.14.

(`munin-plugins-c` doesn't work yet, since that relies on `/proc` which doesn't exist on macOS.  But this PR enables `munin-node-c` to work with other plugins.)

steveschnepp commented:
Nice!

Would you be awesome by adding corresponding lines to the `.travis.yml` file?

Note that I'll squash the PR into only 2 commits prior to merge.

smokris commented:
Sure, I modified the Travis CI configuration to also build on macOS.

steveschnepp commented:
I rewrote most of your commit messages in https://github.com/steveschnepp/munin-c/commits/macos 
If you get those commits in this PR, I'll happily merge the PR ;-)

smokris commented:
OK, I updated the commit messages to match yours, and squashed/dropped the debug commits.

The Travis CI build failure seems to be an issue on their end

steveschnepp commented:
Yep, restarting the build made it pass.

smokris commented:
Great!  Thanks.

steveschnepp commented:
note, i started to port plugins to BSD, but it ended up being a little more complex than anticipated...

steveschnepp commented:
have a look at #47
